### ISSUE TITLE
Disable Root Project From triggering Tag or Release

### ISSUE THREAD

**Issue by scprek** [allegro/axion-release-plugin#694]
Hi,

I apologize for lots of questions, but not sure I saw anywhere else to ask? I'm wondering if I'm using this plugin's monorepo support entirely wrong. 

## Expectations
1. Use plugin's mono repo support to ONLY manage the submodules. Meaning never want any tags or artifacts from the root project. This is not a Platform library with a BOM or anything, just didn't want N repos.
2. Gradle publish only publishes the submodule that changed with the new version (via the prefix support) to Github Packges

If not possible I will either
1. Give up mono repo support or use https://github.com/KengoTODA/gradle-semantic-release-plugin to get auto detection of version increments from commit messages.
2. Make separate Git Repos

## What's Happening

No matter what I've tried, changes in the root directory like the REAMDE or root build.gradle file cause a Git Tag Version bump without the submodules prefix, but still causes ALL the submodules to publish and bump to that non-prefixed version in Github Packages

**HOWEVER** the prefixed submodule Tag versions stay the same. This then causes the odd scenario where the gradle release creates the correct next prefixed tag, but the gradle publish continues with the next version based on the root tag from the step above.

 
<img width="455" alt="Screenshot 2024-01-08 at 10 49 19 AM" src="https://github.com/allegro/axion-release-plugin/assets/2783947/2bc935c7-1242-4313-96b0-6bde7424e3a7">

https://axion-release-plugin.readthedocs.io/en/latest/configuration/version/#tag-with-the-highest-version doesn't seem to fix this, since the root version is technically the highest?

### Setup

<details>
  <summary>build.gradle.kts</summary>
  
```kotlin
plugins {
    id("maven-publish")
    id("jacoco")
    id("pl.allegro.tech.build.axion-release") version "1.16.1"
    id("org.sonarqube") version "4.4.1.3373"
}

scmVersion {
    branchVersionCreator.put("main", "simple") // <-- Tried this
    localOnly.set(true) // <-- Tried this
multiple-versions
    monorepo {
        exclude(project.subprojects.map { it.name })
        // exclude(project.rootProject.name) <-- Tried this, ha
    }
}


subprojects {
    plugins.apply("maven-publish")
    plugins.apply("org.sonarqube")
    plugins.apply("jacoco")
    plugins.apply("pl.allegro.tech.build.axion-release")


    group = "com.org.micronaut"
    version = scmVersion.version

    scmVersion {
        tag {
            prefix.set(project.name)
            versionSeparator.set("_")
        }
    }

    repositories {
        mavenCentral()
    }

    publishing {
        repositories {
            maven {
                name = "orgRepo"
                url = uri("https://maven.pkg.github.com/org/micronaut-commons")
                credentials(PasswordCredentials::class)
            }
        }
    }

    tasks.withType<Test> {
        useJUnitPlatform()
        finalizedBy("jacocoTestReport")
    }

    tasks.withType<JacocoReport> {
        reports {
            xml.required.set(true)
            csv.required.set(false)
            html.required.set(false)
        }
    }

}
```
</details>

<details>
  <summary>GHA Workflow</summary>

```yaml
on:
  workflow_dispatch:
    inputs:
      release_type:
        description: 'Type of semantic release'
        required: true
        default: 'incrementPatch'
        type: choice
        options:
          - incrementPatch
          - incrementMinor
          - incrementMajor

jobs:
  release:
    timeout-minutes: 20
    name: Release
    runs-on: runner
      - name: Gradle Release
        id: release
        uses: gradle/gradle-build-action@v2
        with:
          arguments: release -Prelease.customUsername=${{ github.actor }} -Prelease.customPassword=${{ secrets.GITHUB_TOKEN }} -Prelease.versionIncrementer=${{ inputs.release_type }} -Prelease.disableChecks
      - name: Gradle Publish
        uses: gradle/gradle-build-action@v2
        with:
          arguments: publish -PorgRepoUsername=${{ github.actor }} -PorgRepoPassword=${{ secrets.GITHUB_TOKEN }}
```
</details>

### Example Changes on root Build.gradle and releasing

This results in a new Github Packages metrics with the version `v0.1.4` which I would expect nothing to happen.

<details>
  <summary>gradlew release root changes</summary>
  
```
> Task :release
Creating tag: v0.1.4
Changes made to local repository only

> Task :metrics:verifyRelease
Skipping uncommitted changes check
Skipping ahead of remote check
Skipping snapshot dependencies check

> Task :metrics:release
Working on released version 0.1.0, nothing to release
Pushing all to remote: origin
```

</details>

### Example Changes on subproject and releasing

This results in a new Github Packages metrics, which seems good but with the version `v0.1.5-SNAPSHOT` but the Metrics tag below is `metrics_0.1.1`. Additionally, I don't know why it's adding `-SNAPSHOT` here when everything is commited

<details>
  <summary>gradlew release submodule changes</summary>

```
> Task :release
Working on released version 0.1.4, nothing to release
Changes made to local repository only

> Task :metrics:verifyRelease
Skipping uncommitted changes check
Skipping ahead of remote check
Skipping snapshot dependencies check

> Task :metrics:release
Creating tag: metrics_0.1.1

Pushing all to remote: origin
```

</details>

### Resulting Github Package Versions of Submodule

> Why is my last attempt a `-SNAPSHOT` when the tag for metrics above looks correct?

<img width="249" alt="Screenshot 2024-01-08 at 10 29 37 AM" src="https://github.com/allegro/axion-release-plugin/assets/2783947/fd26745a-60f4-4d43-ab61-baa2c0d5ab79">

**Comment by scprek** [comment]
Turns out I was just mistakenly trying to include the plugin in the root level so I could do all the configuration in 1 spot in the subprojects block

The issue, is that means the top level becomes part of the release.

I moved the following into each build.gradle.kts directly. I don't know if there's a lazyLoad technique where you can configure a thing without applying the plugin like you can do with other tasks

for Example
```kotlin
    tasks.withType<Test> {
        useJUnitPlatform()
        finalizedBy("jacocoTestReport")
    }
```

```kotlin
version = scmVersion.version

scmVersion {
    tag {
        prefix.set(project.name)
        versionSeparator.set("_")
    }

    branchVersionIncrementer.putAll(
        mapOf<String, Any>(
            "micronaut.*" to VersionProperties.Incrementer { c: VersionIncrementerContext -> c.currentVersion.incrementPatchVersion() }
        )
    )
}
```

And now I don't have the root project
```
> Task :feature-flags:currentVersion

Project version: 0.1.0-SNAPSHOT

> Task :metrics:currentVersion

Project version: 0.1.0-SNAPSHOT
```

**Comment by scprek** [comment]
This resolved everything except always being snapshots when published even tho the tag does have it #695 will move discussion there
